external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/internal/fetcher/KafkaRecordsFetchers.java

```diff
@@ -0,0 +1,41 @@
+package org.apache.storm.kafka.spout.internal.fetcher;
+
+import org.apache.kafka.clients.consumer.ConsumerRebalanceListener;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.storm.kafka.spout.KafkaPartitionReader;
+import org.apache.storm.kafka.spout.KafkaPartitionReaders;
+import org.apache.storm.kafka.spout.KafkaSpout;
+import org.apache.storm.kafka.spout.KafkaSpoutConfig;
+import org.apache.storm.task.TopologyContext;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by liurenjie on 19/12/2016.
+ */
+public final class KafkaRecordsFetchers {
+    public static <K, V> KafkaRecordsFetcher<K, V> kafkaRecordsFetcherOf(KafkaSpoutConfig kafkaSpoutConfig,
+                                                                         KafkaConsumer<K, V> consumer,
+                                                                         TopologyContext context,
+                                                                         ConsumerRebalanceListener rebalanceListener) {
+        if (kafkaSpoutConfig.getManualPartitionAssignment()) {
```

Comment:
I think just a flag is easier to use since the user has to write code to configure it otherwise, especially in the case of using flux to configure storm topology. Though your way is more flexible but I don't think we need the flexibility here since it's an internal implementation and the user do not need to know much about that. Let the user assign the implementation class should be used in cases like tuple builder.